api/app/src/external/commonwell/link.ts

```diff
@@ -0,0 +1,181 @@
+import { Person, getId, LOLA } from "@metriport/commonwell-sdk";
+
+import { commonWell, metriportQueryMeta, apiUrl, CW_ID_URL_ENCODED_PREFIX } from "./api";
+import { Link, LinkSource } from "../../routes/medical/schemas/link";
+import { Config } from "../../shared/config";
+
+const rootId = Config.getSystemRootOID();
+
+export const linkPatientToCommonwellPerson = async (
+  personId: string,
+  patientNumber: number
+): Promise<string | undefined> => {
+  const referenceLink = createReferenceLink(patientNumber);
+  try {
+    const link = await commonWell.patientLink(metriportQueryMeta, personId, referenceLink);
+    const networkLinks = await commonWell.getPatientsLinks(
+      metriportQueryMeta,
+      `${patientNumber}${CW_ID_URL_ENCODED_PREFIX}${rootId}`
+    );
+
+    if (networkLinks._embedded && networkLinks._embedded.networkLink?.length) {
+      const lola1Links = networkLinks._embedded.networkLink.filter(
+        link => link.assuranceLevel === LOLA.level_1
+      );
+
+      lola1Links.forEach(async link => {
+        if (link._links?.upgrade?.href) {
+          await commonWell.upgradeOrDowngradeNetworkLink(
+            metriportQueryMeta,
+            link._links?.upgrade?.href
+          );
+        }
+      });
```

Comment:
if there's a bunch of links, we can do the requests in parallel like we do with DAPI provider requests

Reply:
Smart ill do that